src/components/AccountModule/WalletSyncedInfo.js

```diff
@@ -0,0 +1,153 @@
+import React, { useCallback } from 'react'
+import PropTypes from 'prop-types'
+import { GU, Link } from '@aragon/ui'
+import { getAppPath, getPreferencesSearch } from '../../routing'
+import { useWalletBlockNumber } from '../../wallet'
+import {
+  useWalletSyncState,
+  STATUS_CLIENT_CONNECTION_DROPPED,
+  STATUS_CONNECTION_OK,
+  STATUS_MAJOR_NETWORK_SLOWDOWN,
+  STATUS_NETWORK_SYNC_ISSUES,
+  STATUS_TOO_LITTLE_ETH,
+  STATUS_WALLET_CONNECTION_DROPPED,
+} from './utils'
+
+function WalletSyncedInfo({
+  clientListening,
+  clientOnline,
+  clientSyncDelay,
+  locator,
+  walletSyncDelay,
+  walletListening,
+}) {
+  const walletBlockNumber = useWalletBlockNumber()
+  const walletSyncInfo = useWalletSyncState(
+    clientListening,
+    walletListening,
+    clientOnline,
+    clientSyncDelay,
+    walletSyncDelay,
+    walletBlockNumber
+  )
+
+  return (
+    <React.Fragment>
+      {walletSyncInfo.header && (
+        <div
+          css={`
+            margin-top: ${1 * GU}px;
+          `}
+        >
+          <span
+            css={`
+              padding-right: ${1 * GU}px;
+              opacity: 0.8;
+            `}
+          >
+            {walletSyncInfo.header}
+          </span>
+          <span>{walletSyncInfo.info}</span>
+        </div>
+      )}
+      {walletSyncInfo.message !== STATUS_CONNECTION_OK && (
+        <div
+          css={`
+            margin-top: ${1 * GU}px;
+          `}
+        >
+          <ConnectionInfoMessage
+            connectionStatus={walletSyncInfo.message}
+            locator={locator}
+          />
+        </div>
+      )}
+    </React.Fragment>
+  )
+}
+
+WalletSyncedInfo.propTypes = {
+  clientListening: PropTypes.bool,
+  clientOnline: PropTypes.bool,
+  clientSyncDelay: PropTypes.number,
+  locator: PropTypes.object,
+  walletListening: PropTypes.bool,
+  walletSyncDelay: PropTypes.number,
+}
+
+function ConnectionInfoMessage({ connectionStatus, locator }) {
```

Comment:
Let's make sure to fix these linter warnings :).

In this particular case, since this is an internal component, we sometimes disable the linter for prop types but it's also usually not so much effort to just add them in.